### ISSUE TITLE
Error building Docker image with the Caddy module

### ISSUE THREAD

**Issue by FPDK** [protomaps/go-pmtiles#189]
Hi,

When I execute this Dockerfile (Docker v. 4.23.0):

```
ARG CADDY_VERSION=2.8.4

FROM caddy:${CADDY_VERSION}-builder AS builder

RUN xcaddy build --with github.com/protomaps/go-pmtiles/caddy

FROM caddy:${CADDY_VERSION}-alpine

COPY --from=builder /usr/bin/caddy /usr/bin/caddy

# Copy Caddyfile and .pmtiles into the container
COPY Caddyfile .
COPY monaco-latest.pmtiles .

# Expose the port that Caddy will run on
EXPOSE 80
EXPOSE 8080
EXPOSE 443

# Start Caddy
CMD ["caddy", "run", "--config", "Caddyfile"]
```


I get this error during the build step:
```
=> ERROR [builder 2/2] RUN xcaddy build --with github.com/protomaps/go-pmtiles/caddy                                                                                                                110.9s
------                                                                                                                                                                                                      
 > [builder 2/2] RUN xcaddy build --with github.com/protomaps/go-pmtiles/caddy:                                                                                                                             
...
70.96 go: downloading github.com/spaolacci/murmur3 v1.1.0
71.05 2024/10/24 11:06:22 [INFO] exec (timeout=0s): /usr/local/go/bin/go build -o /usr/bin/caddy -ldflags -w -s -trimpath -tags nobadger 
92.94 # go.opentelemetry.io/otel/sdk/trace
92.94 /go/pkg/mod/go.opentelemetry.io/otel/sdk@v1.24.0/trace/span.go:165:20: cannot use (*recordingSpan)(nil) (value of type *recordingSpan) as ReadWriteSpan value in variable declaration: *recordingSpan does not implement ReadWriteSpan (missing method AddLink)
92.94           have addLink("go.opentelemetry.io/otel/trace".Link)
92.94           want AddLink("go.opentelemetry.io/otel/trace".Link)
92.94 /go/pkg/mod/go.opentelemetry.io/otel/sdk@v1.24.0/trace/span.go:804:20: cannot use nonRecordingSpan{} (value of type nonRecordingSpan) as "go.opentelemetry.io/otel/trace".Span value in variable declaration: nonRecordingSpan does not implement "go.opentelemetry.io/otel/trace".Span (missing method AddLink)
92.94 /go/pkg/mod/go.opentelemetry.io/otel/sdk@v1.24.0/trace/tracer.go:50:21: impossible type assertion: p.(*recordingSpan)
92.94   *recordingSpan does not implement "go.opentelemetry.io/otel/trace".Span (missing method AddLink)
92.94           have addLink("go.opentelemetry.io/otel/trace".Link)
92.94           want AddLink("go.opentelemetry.io/otel/trace".Link)
92.94 /go/pkg/mod/go.opentelemetry.io/otel/sdk@v1.24.0/trace/tracer.go:120:10: cannot use tr.newNonRecordingSpan(sc) (value of type nonRecordingSpan) as "go.opentelemetry.io/otel/trace".Span value in return statement: nonRecordingSpan does not implement "go.opentelemetry.io/otel/trace".Span (missing method AddLink)
92.94 /go/pkg/mod/go.opentelemetry.io/otel/sdk@v1.24.0/trace/tracer.go:122:9: cannot use tr.newRecordingSpan(psc, sc, name, samplingResult, config) (value of type *recordingSpan) as "go.opentelemetry.io/otel/trace".Span value in return statement: *recordingSpan does not implement "go.opentelemetry.io/otel/trace".Span (missing method AddLink)
92.94           have addLink("go.opentelemetry.io/otel/trace".Link)
92.94           want AddLink("go.opentelemetry.io/otel/trace".Link)
110.5 2024/10/24 11:07:01 [INFO] Skipping cleanup as requested; leaving folder intact: /tmp/buildenv_2024-10-24-1105.232989868
110.5 2024/10/24 11:07:01 [FATAL] exit status 1
------
Dockerfile:5
--------------------
   3 |     FROM caddy:${CADDY_VERSION}-builder AS builder
   4 |     
   5 | >>> RUN xcaddy build --with github.com/protomaps/go-pmtiles/caddy
   6 |     
   7 |     FROM caddy:${CADDY_VERSION}-alpine
--------------------
ERROR: failed to solve: process "/bin/sh -c xcaddy build --with github.com/protomaps/go-pmtiles/caddy" did not complete successfully: exit code: 1
```

I have attached the full log:
[docker-caddy-log.log](https://github.com/user-attachments/files/17505860/docker-caddy-log.log)


When I don't include the go-pmtiles module (i.e. remove the `--with` argument) the build-step executes without errors.
I have tried building with another custom module (github.com/dunglas/mercure/caddy) and this works fine as well.

Am I doing something wrong?

**Comment by bdon** [comment]
Thanks for the detailed report. Should be fixed on main and in `v1.22.1`, can you try that?

**Comment by FPDK** [comment]
It works fine now. Thank you for the quick response.
